### PR TITLE
Fixes #21575 - Use hound to run eslint

### PR DESCRIPTION
**Pull request by danseethaler**
Adding eslint to .hound.yml, fixing lint errors,
and adding eslint-plugin-prettier for prettier
enforcement.

http://projects.theforeman.org/issues/21575

### CODE REVIEW

File: webpack/index.js [hunk 1]
```diff
@@ -1,3 +1,9 @@
+/* eslint import/no-unresolved: [2, { ignore: [foremanReact/*] }] */
+/* eslint-disable import/no-extraneous-dependencies */
+/* eslint-disable import/extensions */
+
+test
```
[the remark this answers is not in the document]
Comment (danseethaler) [reply webpack/index.js]: I threw in this violation to make sure hound was running on the webpack files.

**Comment by danseethaler** [comment]
Also moved `.eslintrc.js` to `.eslintrc` to simplify and keep in harmony with foreman.

**Comment by danseethaler** [comment]
[test katello]

**Comment by waldenraines** [comment]
[test]

**Comment by waldenraines** [comment]
```
21:14:05 [4mRunning "eslint:target" (eslint) task[24m
21:14:05 [33mWarning: Cannot find module 'eslint-plugin-jest' Use --force to continue.[39m
21:14:05 
21:14:05 [31mAborted due to warnings.[39m
```

**Comment by danseethaler** [comment]
@ehelms can you take a look and see what might be happening here?

The build is failing on the `grunt ci` task for bastion_katello because it can't find the `eslint-plugin-jest` module. This module is included in the top level `.eslintrc` and shouldn't be needed at all in the `grunt ci` task that is linting the `bastion_katello` folder. It feels pretty similar to the issues we ran into when adding the `package.json` to the top level of Katello a few weeks ago.

**Comment by ehelms** [comment]
Looks like eslint is trying to look for `.eslintrc` and finds it a few levels up. Seems like we either need to teach it how to not do this or use the same eslint rules across and run it from a single place. I think the theory is it looks up the directory true to create "composite" rules. So we need to figure how to tell it to stop doing that.

**Comment by danseethaler** [comment]
Good call @ehelms. I didn't realize that eslint went all the way up directory tree until it gets to the root directory looking for `.eslintrc` files. Fortunately we can add a `"root": true` to stop it as specified [in the docs](https://eslint.org/docs/user-guide/configuring#configuration-cascading-and-hierarchy).

**Comment by danseethaler** [comment]
@ehelms looks like it's failing at the same place. Can you confirm that the new `.eslintrc` file under `engines/bastion_katello/` is in the right place? I thought there would already be a `.eslintrc` file there... The `grunt ci` command is running successfully locally.

**Comment by danseethaler** [comment]
[test]

**Comment by waldenraines** [comment]
[test]

**Comment by waldenraines** [comment]
[test]

**Comment by danseethaler** [comment]
@ehelms we've added the `root` flag in the bastion repo but that doesn't seem to fix the issue. Can you take another look?

**Comment by waldenraines** [comment]
[test]

**Comment by ehelms** [comment]
Does this still fail if you put one `.estlintrc` with root configured in `engines/bastion_katello` ?

**Comment by danseethaler** [comment]
@ehelms yeah same error 🤔

**Comment by ehelms** [comment]
[test]

**Comment by ehelms** [comment]
I hacked around the original error for the time being in the job configuration.

**Comment by danseethaler** [comment]
@ehelms hm... the current error is `Cannot read property 'length' of undefined` with no indication of where the error occurred. Any ideas on that?

**Comment by danseethaler** [comment]
Well I had to remove the `babel-eslint` parser which we'll need at some point. Still not sure why the `root` flag isn't stopping `eslint` from looking up the directory chain.

But for now this is a good start. I'm good to merge.

**Comment by ehelms** [comment]
At at least you found the troublesome plugin and we can focus in on why it causes failures as well after this.

**Pull request merged**